test/musica/micm/test_micm_api.F90

```diff
@@ -0,0 +1,122 @@
+  constituents(2,1,1:5) = (/ 0.75_kind_phys, 8.1e-6_kind_phys, 2.42e-17_kind_phys, &
+                          1.15e-5_kind_phys, 6.61e-9_kind_phys /)
+
+  iulog = 6
+
+  call constituent_props(1)%instantiate( &
+    std_name="water_vapor_mixing_ratio",      &
+    long_name="water_vapor_mixing_ratio",     &
+    units="kg kg-1",                          &
+    default_value=0._kind_phys,               &
+    vertical_dim="vertical_layer_dimension",  &
+    advected=.true.,                          &
+    molar_mass=molar_mass_arr(1),             &
+    errcode=errcode, errmsg=errmsg)
+  call constituent_props(2)%instantiate( &
+    std_name="water_vapor_mixing_ratio_a",    &
+    long_name="water_vapor_mixing_ratio",     &
+    units="kg kg-1",                          &
+    default_value=0._kind_phys,               &
+    vertical_dim="vertical_layer_dimension",  &
+    advected=.true.,                          &
+    molar_mass=molar_mass_arr(2),             &
+    errcode=errcode, errmsg=errmsg)
+  call constituent_props(3)%instantiate( &
+    std_name="water_vapor_mixing_ratio_b",    &
+    long_name="water_vapor_mixing_ratio",     &
+    units="kg kg-1",                          &
+    default_value=0._kind_phys,               &
+    vertical_dim="vertical_layer_dimension",  &
+    advected=.true.,                          &
+    molar_mass=molar_mass_arr(3),             &
+    errcode=errcode, errmsg=errmsg)
+  call constituent_props(4)%instantiate( &
+    std_name="water_vapor_mixing_ratio_c",    &
+    long_name="water_vapor_mixing_ratio",     &
+    units="kg kg-1",                          &
+    default_value=0._kind_phys,               &
+    vertical_dim="vertical_layer_dimension",  &
+    advected=.true.,                          &
+    molar_mass=molar_mass_arr(4),             &
+    errcode=errcode, errmsg=errmsg)
+  call constituent_props(5)%instantiate( &
+    std_name="water_vapor_mixing_ratio_d",    &
+    long_name="water_vapor_mixing_ratio",     &
+    units="kg kg-1",                          &
+    default_value=0._kind_phys,               &
+    vertical_dim="vertical_layer_dimension",  &
+    advected=.true.,                          &
+    molar_mass=molar_mass_arr(5),             &
+    errcode=errcode, errmsg=errmsg)
+
+  do i = 1, 5
+    const_prop => constituent_props(i)
+    call constituent_props_ptr(i)%set(const_prop, errcode, errmsg)
+  end do
+
+  call micm_init("chapman", iulog, errcode, errmsg)
+
+  write(*,*) "    -- Initial time_step", time_step
+  write(*,*) "    -- Initial temp", temperature
+  write(*,*) "    -- Initial pressure", pressure
+  write(*,*) "    -- Initial concentrations", constituents
+
+  if (errcode == 0) then
+    call micm_run(time_step, temperature, pressure, dry_air_density, constituent_props_ptr, &
+                constituents, iulog, errcode, errmsg)
+
+    write(*,*) "    -- After solving, conentrations", constituents
```

Comment:
Should we have a check after `micm_run` here that stops the program if the returned `errcode` is non-zero?